File: languages/csharp/01-foundations/arrays-and-vectors/example/vector-filter.cs
// This helper example focuses on isolating one collection transformation so the filtering rule stands out.

// This extra example extends arrays-and-vectors with a focused filtering task.

using System;
using System.Collections.Generic;

// Keep this helper separate so the main example can focus on the larger idea without extra noise.
class Program
{
    static void Main()
    {
        Console.Write("How many numbers? ");
        if (!int.TryParse(Console.ReadLine(), out int count) || count <= 0)
        {
            Console.WriteLine("Please enter a positive count.");
            return;
        }

        List<int> values = new List<int>(count);
        for (int index = 0; index < count; index++)
        {
            Console.Write($"Value {index + 1}: ");
            if (!int.TryParse(Console.ReadLine(), out int value))
            {
                Console.WriteLine("Please enter valid integers only.");
                return;
            }

            values.Add(value);
        }

        Console.Write("Filter values greater than or equal to: ");
        if (!int.TryParse(Console.ReadLine(), out int threshold))
        {
            Console.WriteLine("Please enter a valid threshold.");
            return;
        }

        List<int> filtered = values.FindAll(value => value >= threshold);
        if (filtered.Count > 0)
        {
            Console.WriteLine("Filtered values: " + string.Join(", ", filtered));
        }
        else
        {
            Console.WriteLine("Filtered values: none");
        }
    }
}

File: languages/go/01-foundations/arrays-and-vectors/example/vector-filter.go
// This helper example focuses on isolating one collection transformation so the filtering rule stands out.

// This extra example extends arrays-and-vectors with a focused filtering task.

package main

import "fmt"

// Keep this helper separate so the main example can focus on the larger idea without extra noise.
func main() {
	var count int
	fmt.Print("How many numbers? ")
	if _, err := fmt.Scan(&count); err != nil || count <= 0 {
		fmt.Println("Please enter a positive count.")
		return
	}

	values := make([]int, 0, count)
	for index := 0; index < count; index++ {
		var value int
		fmt.Printf("Value %d: ", index+1)
		if _, err := fmt.Scan(&value); err != nil {
			fmt.Println("Please enter valid integers only.")
			return
		}
		values = append(values, value)
	}

	var threshold int
	fmt.Print("Filter values greater than or equal to: ")
	if _, err := fmt.Scan(&threshold); err != nil {
		fmt.Println("Please enter a valid threshold.")
		return
	}

	fmt.Print("Filtered values: ")
	printedAny := false
	for _, value := range values {
		if value >= threshold {
			if printedAny {
				fmt.Print(", ")
			}
			fmt.Print(value)
			printedAny = true
		}
	}

	if !printedAny {
		fmt.Print("none")
	}
	fmt.Println()
}
